File: Two-Pointers/Easy/283-Move-Zeroes/Solution.py
class Solution:
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        left, right = 0, 0
        while right < len(nums):
            if nums[right] != 0:
                if left != right:
                    nums[left] = nums[right]
                left += 1
            right += 1
            
        while left < len(nums):
            if nums[left] != 0:
                nums[left] = 0
            left += 1

File: Two-Pointers/Easy/283-Move-Zeroes/Solution.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        // two pointers: same direction
        // tc:O(n); sc:O(1)
        if(nums == null || nums.Length < 2) { //corner case
            return;
        }
        int slow = 0, fast = 0; //slow points to 0 elem; fast points to non-0 elem
        while(fast < nums.Length) {
            if(nums[fast] != 0) {
                if(slow != fast) { // avoid wrost case 1,1,1,1,0
                    nums[slow] = nums[fast];
                }
                slow++; 
            }
            fast++;
        }
        
        while(slow < nums.Length) { // assign 0 to end elems
            if(nums[slow] != 0) {
                nums[slow] = 0;
                slow++;
            }
        }
    }
}